shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java

```diff
@@ -0,0 +1,647 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.embedding.android;
+
+import android.app.Activity;
+import android.arch.lifecycle.Lifecycle;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+import android.os.Handler;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentActivity;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import java.util.Arrays;
+
+import io.flutter.Log;
+import io.flutter.app.FlutterActivity;
+import io.flutter.embedding.engine.FlutterEngine;
+import io.flutter.embedding.engine.FlutterShellArgs;
+import io.flutter.embedding.engine.dart.DartExecutor;
+import io.flutter.embedding.engine.renderer.OnFirstFrameRenderedListener;
+import io.flutter.plugin.platform.PlatformPlugin;
+import io.flutter.view.FlutterMain;
+
+import static android.content.ComponentCallbacks2.TRIM_MEMORY_RUNNING_LOW;
+
+/**
+ * Delegate that implements all Flutter logic that is the same between a {@link FlutterActivity}
+ * and a {@link FlutterFragment}.
+ * <p>
+ * <strong>Why does this class exist?</strong>
+ * <p>
+ * One might ask why an {@code Activity} and {@code Fragment} delegate needs to exist. Given
+ * that a {@code Fragment} can be placed within an {@code Activity}, it would make more sense
+ * to use a {@link FlutterFragment} within a {@link FlutterActivity}. Then, the problem of
+ * duplicated behavior is solved without extracting all of that logic into a 3rd party
+ * {@code Object}.
+ * <p>
+ * The original solution for this embedding was exactly that - a {@link FlutterFragment}
```

Comment:
There's too much narration. Just state bulletpoints :)

Reply:
I'm not sure why bullet points would help, but I reduced the verbosity.

Reply:
I didn't mean literal bullet points. just bullet point verbosity